src/transformers/tokenization_utils.py

```diff
@@ -495,46 +516,48 @@ def tokenize(self, text: TextInput, **kwargs) -> List[str]:
             `List[str]`: The list of tokens.
         """
         # Simple mapping string => AddedToken for special tokens with specific tokenization behaviors
-        all_special_tokens_extended = {
-            str(t): t for t in self.all_special_tokens_extended if isinstance(t, AddedToken)
-        }
-
         text, kwargs = self.prepare_for_tokenization(text, **kwargs)
 
         if kwargs:
             logger.warning(f"Keyword arguments {kwargs} not recognized.")
 
-        # TODO: should this be in the base class?
         if hasattr(self, "do_lower_case") and self.do_lower_case:
             # convert non-special tokens to lowercase
-            escaped_special_toks = [
-                re.escape(s_tok) for s_tok in (self.unique_no_split_tokens + self.all_special_tokens)
-            ]
+            escaped_special_toks = [re.escape(s_tok) for s_tok in (self.all_special_tokens)]
```

Comment:
This is kind of a hard choice, but `self.unique_no_split_tokens` was nuked. Still debatable but `unique_no_split_tokens` was a way to give users some freedom that we should probably not give. tokens that are added will never be split, and tokens that are part of the vocab.... should never be split either no? + this is mixing being a special tokens (skipped when decoding, and not normalized)  and unique no split tokens